src/connectors/search-box/connectSearchBox.js

```diff
@@ -139,6 +133,19 @@ export default function connectSearchBox(renderFn, unmountFn = noop) {
         return state.setQueryParameter('query', undefined);
       },
 
+      getWidgetRenderState(renderState, { helper, searchMetadata }) {
+        return {
+          ...renderState,
+          searchBox: {
```

Comment:
I know we agreed on spreading for consistency, but now it seems a little odd to me. I wonder if we could have the key being calculated based on the $$type and return directly the object containing query? Not sure yet what's better, but this seems quite indirect

Reply:
The reason I'm thinking this, is because the hook is get**Widget**RenderState, not `getCurrentRenderState`

Reply:
> I wonder if we could have the key being calculated based on the $$type

That's what I was thinking initially but in the end it's way more shady. That's an implicit API.

> The reason I'm thinking this, is because the hook is get**Widget**RenderState, not `getCurrentRenderState`

We've got the same behavior for `getWidgetUiState` which returns the whole index UI state, not the widget UI state.